package/MDAnalysis/core/groups.py

```diff
@@ -850,6 +850,128 @@ def center_of_geometry(self, pbc=None, compound='group'):
         return self.center(None, pbc=pbc, compound=compound)
 
     centroid = center_of_geometry
+    
+    @warn_if_not_unique
+    def accumulate(self, attribute, op=np.sum, compound='group'):
+        """Accumulates the attribute associated with (compounds of) the group.
+        
+        Accumulates the attribute of :class:`Atoms<Atom>` in the group.
+        The accumulation is done per :class:`Residue`, :class:`Segment`, molecule, 
+        or fragment can be obtained by setting the `compound` parameter
+        accordingly. By default the method sums up all elements, but 
+        any numpy operation applying to numpy.ndarray can be used.
+
+        Parameters
+        ----------
+        attribute : str
+            Attribute to accumulate.
+        op : operation, optional
+            Operation to perform.
+        compound : {'group', 'segments', 'residues', 'molecules', 'fragments'}, optional
+            If ``'group'``, the sum over all elements associated
+            with atoms in the group will
+            be returned as a single value. Else, the sum for all elements
+            of each :class:`Segment`, :class:`Residue`, molecule, or fragment
+            will be returned as an 1d array.
+            Note that, in any case, *only* the positions of :class:`Atoms<Atom>`
+            *belonging to the group* will be taken into account.
+
+        Returns
+        -------
+        sum : numpy.ndarray
```

Comment:
You do not have to provide a name for the return value. Here I would avoid doing it as the name is actually wrong:

```
Returns
-------
numpy.ndarray
    The description
```